SimpleExec/ProcessExtensions.cs

```diff
@@ -15,34 +15,18 @@ public static void Run(this Process process, bool noEcho, string echoPrefix)
 
         public static async Task RunAsync(this Process process, bool noEcho, string echoPrefix, CancellationToken cancellationToken)
         {
-            var tcs = new TaskCompletionSource<object>();
-
-            using (var started = new SemaphoreSlim(0, 1))
-            using (var ranToCompletionOrCanceled = new SemaphoreSlim(1, 1))
-            using (var cancellation = cancellationToken.Register(() =>
+            var tcs = new TaskCompletionSource<object>(TaskCreationOptions.RunContinuationsAsynchronously);

```

Comment:
This should be the default but it isn't because backward compatibility. If not set the continuation might complete on the thread that triggered TrySetResult or TrySetCanceled. This is almost never the desired behaviour unless you need to do some weird perf optimization and you know completing on the thread that triggered will never lead to deadlock scenarios

Reply:
👍 thanks, makes sense.